clicommand/pipeline_upload.go

```diff
@@ -107,6 +117,17 @@ var PipelineUploadCommand = cli.Command{
 			Usage:  "When true, fail the pipeline upload early if the pipeline contains secrets",
 			EnvVar: "BUILDKITE_AGENT_PIPELINE_UPLOAD_REJECT_SECRETS",
 		},
+		cli.StringFlag{
+			Name:   "signing-key",
+			Usage:  "Path to a file containing a signing key. Passing this flag enables pipeline signing. The format of the file depends on the chosen signing algorithm",
+			EnvVar: "BUILDKITE_PIPELINE_SIGNING_KEY_FILE",
+		},
```

Comment:
a couple questions on this:
- for usability, we might wanna call this `--signing-key-file` (or `--signing-key-path`, which seems to be more standard in the agent? feels clunky to me though)
- do we want to support passing a signing key into the cli directly? eg `--signing-key "123deadbeef321"`
- the envar should probably be `BUILDKITE_PIPELINE_UPLOAD_SIGNING_KEY_FILE` (added `UPLOAD`)

Reply:
- Consistency sounds good
- One danger with passing keys via flag is that keys could end up in shell history or process info (`ps -x`), but maybe that's not a problem?
- Sounds good